generate_images.js

```diff
@@ -71,20 +76,75 @@ var leeroy = {
 	"playerClass": "NEUTRAL",
 };
 
-function drawCard(card) {
-	sunwell.init()
-	sunwell.createCard(leeroy, 512, function(canvas) {			
-		var out = fs.createWriteStream(__dirname + '/card' + card.gameId + '.png')
-		var stream = canvas.pngStream();
+function drawAllCards(json, dir) {
+	for (var i = 0; i < Math.min(json.length, 1000000); i++) {
+		var c = json[i];
+		c.gameId = c.id;
 		
-		stream.on('data', function(chunk){
-			out.write(chunk);
-		});
+		if (true) {
+			var fileName = dir + '/card_' + c.gameId + '.png';
+			if (!c.type || !c.playerClass) {
+				console.log("skip " + fileName);
+			} else if (!fs.existsSync(__dirname + "/" + sunwell.settings.textureFolder + "/" + c.gameId + '.jpg')) {
+				console.log("skip " + fileName);
+			} else if (!fs.existsSync(fileName)) {
+				console.log("queue " + fileName);
+				(function(c, i, fileName) {
+					sunwell.createCard(c, 512, function(canvas) {
+						var out = fs.createWriteStream(fileName)
+						var stream = canvas.pngStream();
+						
+						stream.on('error', function(chunk){
+						  console.log("ouille");
+						});
 
-		stream.on('end', function(){
-			console.log('saved png');
-		});
+						stream.on('data', function(chunk){
+							out.write(chunk);
+						});
+
+						stream.on('end', function(){
+							console.log("saved " + i + "/" + json.length + ": " + fileName);
+							out.end();
+							
+							if (i == json.length - 1) {
+								console.log("done");
+							}
+						});
+					});
+				})(c, i, fileName);
+			} else {
+				//console.log("skip existing" + fileName);
+			}
+		}
+	}
+}
+
+
+function generateLang(lang) {
+	var dir = outputDirectory + "/" + lang;
+	
+	function downloadJson(url) {
+		request(url, function(error, response, body) {
+			if (!error && response.statusCode === 200) {
+				const j = JSON.parse(body)				
+				drawAllCards(j, dir);
+			} else {
+				console.log("Got an error: ", error, ", status code: ", response.statusCode)
+			}
+		})
+	}
+	mkdirp(dir, function(err) {
+		downloadJson("https://api.hearthstonejson.com/v1/15590/" + lang + "/cards.json");
```

Comment:
Can you make this a global/configurable somehow? You can also use `latest` instead of  a build to always get the latest one.